pkg/reconciler/managed/reconciler.go

```diff
@@ -662,6 +664,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"external-name", meta.GetExternalName(managed),
 	)
 
+	// check the pause annotation and return if it has the value "true"
+	// after logging, publishing an event and updating the SYNC status condition
+	if managed.GetAnnotations()[meta.AnnotationKeyReconciliationPaused] == "true" {
+		log.Debug("Reconciliation is paused via the pause annotation", "annotation", meta.AnnotationKeyReconciliationPaused, "value", "true")
+		record.Event(managed, event.Normal(reasonReconciliationPaused, "Reconciliation is paused via the pause annotation"))
+		managed.SetConditions(xpv1.ReconcilePaused())
```

Comment:
We should either remove the condition or set it to another value once the annotation is removed, which could be right after this condition block.

Reply:
Looks like if the reconciliation is not paused, `Synced` status condition either stays at the `False` state but this time with the reason `ReconcileError` or transitions to the state `True` with reason `ReconcileSuccess`. So the idea is if the reconciliation is paused via the annotation, we have `Synced` condition set to `False` with reason `ReconcilePaused`. We stay at this state (condition) as long as the annotation is set to `true`. Then if the annotation's value changes, then the managed resource acquires either the `Synced` condition set to `True` with the reason `ReconcileSuccess` or again the `Synced` condition set to `False` but this time with the reason `ReconcileError`. Do you think there is a possibility this transition would not happen (apart from a failing status update)?

Reply:
Oh OK, sorry, I thought this was a separate condition. All good, thank you for explanation!